### PR TITLE
Closes #344: added ovirt_disk_resize resource

### LINKED ISSUE #344: Implement the `ovirt_disk_resize` resource
## Please describe what you would like to see

We need a resource that can resize disks to a specified size.

## Please describe the solution you'd like

The resource should look as follows:

```terraform
resource "ovirt_disk_size" "master" {
  disk_id = var.disk_id
  size      = var.disk_size_in_bytes
}
```

The operation should make sure the disk size is as specified when the resource is applied.

## Please describe your use case

We need to resize the disks after a VM is created from a template. See openshift/installer#5867 for details.

### PR DESCRIPTION
**Pull request by engelmi**
## Please describe the change you are making

This PR fixes #344 by introducing the ovirt_disk_resize resource. 

## Are you the owner of the code you are sending in, or do you have permission of the owner?

yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

yes

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: ovirt/resource_ovirt_disk_resize.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		}
+	}
+
+	updateFailedDiag := diag.Diagnostic{
+		Severity: diag.Error,
+		Summary:  "Failed to update disk size.",
+	}
+	diskUpdate, err := client.StartUpdateDisk(ovirtclient.DiskID(diskID), params)
+	if err != nil {
+		updateFailedDiag.Detail = err.Error()
```
Comment (janosdebugs) [inline ovirt/resource_ovirt_disk_resize.go:99]: You need to check for not found errors and set the ID to empty in these cases.

File: ovirt/resource_ovirt_disk_resize.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		return diag.Diagnostics{updateFailedDiag}
+	}
+	_, err = diskUpdate.Wait()
+	if err != nil {
+		updateFailedDiag.Detail = err.Error()
```
Comment (janosdebugs) [inline ovirt/resource_ovirt_disk_resize.go:104]: Same here.

**Review by janosdebugs — CHANGES_REQUESTED** [review]
Please check for not found errors.

File: ovirt/resource_ovirt_disk_resize.go (hunk 1 above)
Comment (engelmi) [reply ovirt/resource_ovirt_disk_resize.go:99]: Good point! Will add the not found check.

File: ovirt/resource_ovirt_disk_resize.go (hunk 2 above)
Comment (engelmi) [reply ovirt/resource_ovirt_disk_resize.go:104]: Doesn't the not found error only happen in `StartUpdateDisk` and is handled then?

Comment (engelmi) [reply ovirt/resource_ovirt_disk_resize.go:104]: True, I'll add the check.

**Review by janosdebugs — APPROVED** [review]

**Pull request closed without merging**